accepted/targeting-packs-and-runtime-packs.md

```diff
@@ -0,0 +1,287 @@
+
+* .NET Core SDK 3.0.100 with 3.0.0 runtime
+    * packs/Microsoft.NETCore.App.Ref/3.0.0
+    * shared/Microsoft.NETCore.App/3.0.0
+
+* .NET Core SDK 3.0.101 with 3.0.1 runtime
+    * packs/Microsoft.NETCore.App.Ref/3.0.0
+    * shared/Microsoft.NETCore.App/3.0.1
+
+
+## Package Layout
+
+Since the packages will have a special type and not be installable into projects in the usual way, we do not need to use folders in the nupkg like `ref/\<TFM>` etc. Instead we will simply place the pack contents in the root of the .nupkg. The package names and versions already uniquely identify the content per TFM/RID and a big part of the package size savings comes from avoiding assets that span multiple TFMs in a single package.
+
+
+# FrameworkReference
+
+A `FrameworkReference` is a new MSBuild item that represents a reference to a well-known *group* of framework assemblies that are versioned with the project's `TargetFramework`.
+
+For .NET Core, there are two use cases:
+
+* Reference to entire shared framework
+``` xml
+ <FrameworkReference Include="Microsoft.NETCore.App" />
+ <FrameworkReference Include="Microsoft.AspNetCore.App" />
+ ```
+
+* Reference to named subset or "profile" of a shared framework
+``` xml
+<FrameworkReference Include="Microsoft.WindowsDesktop.App|WindowsForms" />
+<FrameworkReference Include="Microsoft.WindowsDesktop.App|WPF" />
+```
+
+The `<shared framework>|<profile name>` is not final, and we may decide to simply assign a unique full name to each profile.
+
+A FrameworkReference can be added to a project in the following ways.
+
+  1. Implicit `<FrameworkReference Include="...">` in MSBuild SDK.
+  2. Transitively through project references
+  3. Transitively through package references
+  3. Explicit `<FrameworkReference Include="..." />` in user project file
+
+The implicit framework references broken down by MSBuild SDK for netcoreapp3.0 are as follows:
+
+ 1. Microsoft.NET.Sdk 
+    * "Microsoft.NETCore.App"
+ 
+ 2. Microsoft.NET.Sdk.Web
+    * "Microsoft.AspNetCore.App" 
+    * "Microsoft.NETCore.App" (via chained Microsoft.NET.Sdk)
+ 
+ 3. Microsoft.NET.Sdk.WindowsDesktop
+    * "Microsoft.WindowsDesktop.App" if both $(UseWPF) and $(UseWindowsForms) are true
+    * "Microsoft.WindowsDesktop.App|WPF if only $(UseWPF) is true
+    * "Microsoft.WindowsDesktop.App|WindowsForms" if only $(UseWindowsForms) is true
+    * "Microsoft.NETCore.App" (via chained Microsoft.NET.Sdk)
+
+Just as with the older framework package reference, `$(DisableImplicitFrameworkReferences)` will be honored and prevent the SDKs from adding these implicit FrameworkReferences. There will also be a `$(DisableTransitiveFrameworkReferences)`, which would mirror existing `$(DisableTransitiveProjectReferences)`.
+
+Transitivity will be achieved via recording when a `ProjectReference` or `PackageReference` needs a `FrameworkReference` in the assets file during NuGet restore: https://github.com/NuGet/Home/issues/7342
+
```

Comment:
Question: is it possible/desirable for SDKs or other things to build smarts based on a framework reference?

For instance in Razor we have this general problem to solve - how do we know what runtime your build output targets? Do you target AspNetCore+Mvc? What version?

We'd prefer to tie this kind of decision to something the user already cares about so that it doesn't become noise in the project file. It's not something the user cares to configure, there is only really one correct answer and that answer depends on where your code is going to run.

Is this an approach that you'd recommend investigating?

Reply:
/cc @pranavkm

Reply:
@nguerrera is there an issue tracking adding support for `FrameworkReference` to `Microsoft.Extensions.DependencyModel`?